apsuite/commisslib/meas_bo_tune_by_bpm.py

```diff
@@ -107,30 +107,45 @@ def get_orbit_data(self, injection=False, external_trigger=False):
         if ret:
             trigbpm.delay = self.delay0
             self.data = dict()
-            raise AssertionError(f'Problem waiting BPMs update. Error code: {ret:d}')
+            raise AssertionError(
+                f'Problem waiting BPMs update. Error code: {ret:d}')
         orbx, orby = bobpms.get_mturn_orbit()
         bobpms.cmd_mturn_acq_abort()
         trigbpm.delay = self.delay0
 
+        self.data['orbx'], self.data['orby'] = orbx, orby
+        self.data['timestamp'] = _time.time()
+
+    def get_data(self, delta=''):
+        """."""
+        if not hasattr(self, 'data'):
+            self.data = dict()
+
         # Store data
+        bobpms = self.devices['bobpms']
+        trigbpm = self.devices['trigbpm']
+
         bpm0 = bobpms[0]
         csbpm = bpm0.csdata
         data = dict()
-        data['timestamp'] = _time.time()
+        data['delta'] = delta
         data['rf_frequency'] = self.devices['rfgen'].frequency
         data['current_150mev'] = self.devices['currinfo'].current150mev
         data['current_1gev'] = self.devices['currinfo'].current1gev
         data['current_2gev'] = self.devices['currinfo'].current2gev
         data['current_3gev'] = self.devices['currinfo'].current3gev
-        data['orbx'], data['orby'] = orbx, orby
         data['mt_acq_rate'] = csbpm.AcqChan._fields[bpm0.acq_channel]
         data['bpms_nrsamples_pre'] = bpm0.acq_nrsamples_pre
         data['bpms_nrsamples_post'] = bpm0.acq_nrsamples_post
         data['bpms_trig_delay_raw'] = trigbpm.delay_raw
         data['bpms_switching_mode'] = csbpm.SwModes._fields[
                                         bpm0.switching_mode]
-        data['init_delay'] = self.delay0
-        self.data = data
+        data['qf_wave_form'] = PV('BO-Fam:PS-QF:Wfm-RB').get()
+        data['qd_wave_form'] = PV('BO-Fam:PS-QD:Wfm-RB').get()
+        if hasattr(self, 'delay0'):
```

Comment:
it is not a good practice to have a `delay0` attribute of the measurement class. If this variable reflects a parameter of the measurement, you could add it to the `Params` object, if it is a measured value, store it in the `data` dictionary. If it is a temporary variable, keep it as temporary.

Reply:
I leave it as a temporary variable. Thanks for the tip.